tests/scripts/thread-cert/pktverify/addrs.py

```diff
@@ -0,0 +1,163 @@
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+import ipaddress
+from typing import Union
+
+from pktverify.bytes import Bytes
+
+
+class EthAddr(Bytes):
+
+    def __init__(self, addr: Union[str, bytearray, 'Bytes']):
+        super(EthAddr, self).__init__(addr)
+        if len(self) not in (6, 8):
+            raise ValueError((addr, self))
+
+    def __repr__(self):
+        return '%s(%r)' % (self.__class__.__name__, self.format_octets())
+
+    def __str__(self):
+        return self.format_octets()
+
+
+class ExtAddr(Bytes):
+
+    def __init__(self, addr: Union[str, bytearray, 'Bytes']):
+        super(ExtAddr, self).__init__(addr)
+        if len(self) != 8:
+            raise ValueError((addr, self))
+
+    def __repr__(self):
+        return '%s(%r)' % (self.__class__.__name__, self.format_octets())
+
+    def __str__(self):
+        return self.format_octets()
+
+
+class Ipv6Addr(Bytes):
+
+    def __init__(self, addr: Union[str, bytearray, 'Bytes']):
+        if isinstance(addr, str):
+            # try to parse compacted ipv6 address
+            try:
+                addr = Ipv6Addr._expand(addr)
+            except ipaddress.AddressValueError:
+                pass
+
+        super(Ipv6Addr, self).__init__(addr)
+        if len(self) != 16:
+            raise ValueError((addr, self))
+        self._addr = ipaddress.IPv6Address(self)
+
+    def __repr__(self):
+        return '%s(%r)' % (self.__class__.__name__, self.format_hextets())
+
+    def __str__(self):
+        return self.format_hextets()
+
+    @staticmethod
+    def _expand(addr) -> str:
+        assert isinstance(addr, str)
+        a = ipaddress.IPv6Address(addr)
+        return a.exploded
+
+    @property
+    def is_global(self) -> bool:
+        if self._addr.is_global:
+            return True
+
+        if self._addr.is_link_local or self._addr.is_multicast or self._addr.is_loopback:
+            return False
+
+        return True
+
+    @property
+    def is_dua(self) -> bool:
+        from pktverify import consts
+        return self.startswith(consts.DOMAIN_PREFIX)
+
+    @property
+    def is_backbone(self) -> bool:
+        from pktverify import consts
+        return self.startswith(consts.BACKBONE_IPV6_PREFIX)
+
+    @property
+    def is_link_local(self) -> bool:
+        return self._addr.is_link_local
+
+    @property
+    def is_multicast(self) -> bool:
+        return self._addr.is_multicast
+
+    @property
+    def is_mleid(self) -> bool:
+        from pktverify.consts import DEFAULT_MESH_LOCAL_PREFIX
+        return self.startswith(DEFAULT_MESH_LOCAL_PREFIX)
+
+
+if __name__ == '__main__':
+    a = EthAddr("010203040506")
+    assert a == EthAddr("01:02:03:04:05:06")
+    assert a == EthAddr("0102:0304:0506")
+    assert eval(repr(a)) == a
+    assert a == str(a)
+    assert str(a) == a
+
+    print(a, repr(a))
+    assert isinstance(a[:], Bytes)
+    assert a[:3] == "010203"
+    assert a[3:] == "040506"
```

Comment:
What about merging the two lines into one?

Reply:
Should be just fine since it's a test script.